src/Site.elm

```diff
@@ -71,6 +72,11 @@ pathFromString srcString =
     Pages.Url.fromPath <| Path.fromString srcString
 
 
+fullPathFromString : String -> Pages.Url.Url
+fullPathFromString srcString =
+    Pages.Url.external (String.join "" [ chompTrailingUrlSlash canonicalUrl, srcString ])
```

Comment:
```suggestion
absoluteUrl : String -> Pages.Url.Url
absoluteUrl srcString =
    srcString
        |> Path.fromString
        |> Pages.Url.fromPath
        |> Pages.Url.toAbsoluteUrl canonicalUrl
        |> Pages.Url.external
```
I also came up with this which made sense based on the package types and the functions but it feels like its going round the houses a bit just to join two strings.